cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/io/cdap/cdap/etl/batch/mapreduce/MapReducePreparer.java

```diff
@@ -133,6 +133,12 @@ protected SubmitterPlugin createSource(BatchConfigurable<BatchSourceContext> bat
     });
   }
 
+  @Nullable
+  @Override
+  protected SubmitterPlugin createStreamingSource(StreamingSource<?> streamingSource, StageSpec stageSpec) {
```

Comment:
shouldn't add this method, we don't want spark and mapreduce dependencies to mix. The PipelinePhasePreparer has a catch-all create() method that can be used.